libs/libc/modlib/modlib_loadhdrs.c

```diff
@@ -116,7 +116,6 @@ int modlib_loadhdrs(FAR struct mod_loadinfo_s *loadinfo)
       loadinfo->phdr = (FAR Elf_Phdr *)lib_malloc(phdrsize);
       if (!loadinfo->phdr)
         {
-          lib_free(loadinfo->shdr);
```

Comment:
keep the free call  and assigning the pointer to null

Reply:
@anchao The reason why I removed it is because there is no free action in lines 99, 111 of this file, Looks like modlib_unload is handled better. ditto.

Reply:
1. modlib_loadhdrs() is a public function. All resources are better managed inside the function. If the call fails from internal, it is better to destroy by this function.
2. I don't think delay free to modlib_unload() is a good practice, it will increase the lifetime of the memory being occupied

Reply:
Yes, resources should be released internally, but I look at the usage of libc/dlfcn and sched/module. All resource releases are placed in modlib_uninitialize and modlib_unload. It seems that we can make less changes and reuse some code.